docs/terminology.md

```diff
@@ -0,0 +1,26 @@
+Betting/Protocol Terminology
+============================
+
+## Traditional betting terminology
+
+A 'bet' (noun) is an pledge between to parties (bettors) on the outcome of an event. There are two 'sides' to a bet, the success outcome and the unsuccess outcome.
+Technically a bettor 'lays' on the unsuccessful outcome and 'bets' on the successful outcome, however these two terms are not widely known and common parlance uses the term 'bet' or 'wager' for either success or unsuccess.
+
+## Gun protocol terminology
+
+A bet is between two bettors, Alice and Bob, as the name implies Alice acts first.
+
+1. Alice proposes the desire to be involved in a bet (locks in Alice's wager amount and the event).
```

Comment:
s/proposes/announces/ would work better in this sentence. The slight downside is we already use that term with respect to the oracle "announcing" its intention to attest to an event.

Also should mention how awkward this is and why it's like that. In the future we could allow the proposal to have more stuff in it if the proposer doesn't mind losing some privacy. We could even increase privacy by not having the event at all in there are letting "Bob" choose the event.

Going even crazier it is possible to do an even more covert protocol where the "proposal" would only have a public key and the offerer would choose everything but sign the tx with SIGHASH_SINGLE|ANYONECANPAY. I didn't implement this because it's a bit dangerous to because the offerer MUST cancel their bet even if the proposer takes another offer (which atm naturally cancels the offer).

After writing this I really think I prefer "announcing" a bet rather than proposing one.

I think this doc should actually be in `bet.md` at the start.
I have a mild preference for avoiding "Alice" and "Bob" and sticking to "announcer" and "offerer".

Reply:
After trying the rename I actually prefer proposal over announcement because you can cancel a proposal but it doesn't make sense to talk about cancelling a proposal.

Reply:
Do you want to just take what you like from this PR and put it in `bet.md`? I'm not precious about it, you're welcome to use and abuse it as you see fit :) Otherwise, if you leave this open I'm happy to play around with the wording again when I'm looking at `gun` next.

English is painful, huh.

Cheers

Reply:
Alright, I had another go. Lots of force pushing to check the rendering. See what you think, feel free to make changes without consulting me and merge/close as you see fit.